edgelet/edgelet-core/src/virtualization.rs

```diff
@@ -0,0 +1,20 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+use crate::error::{Error, ErrorKind};
+use failure::ResultExt;
+use std::process::Command;
+
+pub fn is_virtualized_env() -> Result<Option<bool>, Error> {
+    if cfg!(target_os = "linux") {
```

Comment:
I would do this check at compile time. Make 2 functions, one with #[cfg(unix)] and one with #[cfg(windows)]. The windows only returns Ok(None), and the linux function can remove the if.

Reply:
I understood your point but we are not planning to run this function on Windows OS.